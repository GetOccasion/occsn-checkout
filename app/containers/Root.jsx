import React, { Component } from 'react';

import occsn from '../libs/Occasion.js';

import '../styles/index.css'

export default class Root extends Component {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    this.setState({ product: occsn.Product.build({ title: 'Nick' }) });
  }

  render() {
    const { product } =  this.state;

    if(!product) return;

    return (
      <p>Hello { product.title }!!</p>
    );
  }
}
