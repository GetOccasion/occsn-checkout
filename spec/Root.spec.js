import React from 'react';
import Root from '../app/containers/Root.jsx';

describe('MyComponent', () => {
  it('should render my component', () => {
    const wrapper = shallow(<Root />);

    expect(wrapper).toMatchSnapshot();
  });
});
