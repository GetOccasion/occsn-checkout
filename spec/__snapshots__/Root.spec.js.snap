// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render my component 1`] = `
<p>
  Hello World!!
</p>
`;
